File: assets/images/parse_content.js
const fs = require('fs');
const path = require('path');

const filePath = "C:\\Users\\EikoMotsu\\.gemini\\antigravity-ide\\brain\\fd780298-5d85-427b-8f62-af4682affc59\\.system_generated\\steps\\52\\content.md";

try {
  let content = fs.readFileSync(filePath, 'utf8');
  
  // Extract Headings
  console.log("=== HEADINGS ===");
  const headingRegex = /<h([1-6])[^>]*>([\s\S]*?)<\/h\1>/gi;
  let match;
  while ((match = headingRegex.exec(content)) !== null) {
    const level = match[1];
    const text = match[2].replace(/<[^>]*>/g, '').trim();
    if (text) {
      console.log(`H${level}: ${text}`);
    }
  }

  // Extract Images
  console.log("\n=== IMAGES ===");
  const imgRegex = /<img[^>]+src="([^"]+)"[^>]*>/gi;
  while ((match = imgRegex.exec(content)) !== null) {
    console.log(`IMG: ${match[1]}`);
  }

  // Extract Links
  console.log("\n=== LINKS ===");
  const linkRegex = /<a[^>]+href="([^"]+)"[^>]*>([\s\S]*?)<\/a>/gi;
  while ((match = linkRegex.exec(content)) !== null) {
    const href = match[1];
    const text = match[2].replace(/<[^>]*>/g, '').trim();
    if (text) {
      console.log(`LINK: ${text} -> ${href}`);
    }
  }
} catch (err) {
  console.error("Error reading file:", err);
}

File: assets/images/parse_content.py
import re
from bs4 import BeautifulSoup

file_path = r"C:\Users\EikoMotsu\.gemini\antigravity-ide\brain\fd780298-5d85-427b-8f62-af4682affc59\.system_generated\steps\52\content.md"

with open(file_path, "r", encoding="utf-8") as f:
    html_content = f.read()

# Clean metadata lines at the top
html_content = re.sub(r"^Title:.*?\n\n", "", html_content, flags=re.DOTALL)
html_content = re.sub(r"^Description:.*?\n\n", "", html_content, flags=re.DOTALL)
html_content = re.sub(r"^Source:.*?\n\n", "", html_content, flags=re.DOTALL)
html_content = re.sub(r"^---.*?\n\n", "", html_content, flags=re.DOTALL)

soup = BeautifulSoup(html_content, 'html.parser')

# Get all text
print("--- HEADINGS ---")
for h in soup.find_all(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']):
    print(f"{h.name}: {h.get_text().strip()}")

print("\n--- IMAGES ---")
for img in soup.find_all('img'):
    print(f"SRC: {img.get('src') or img.get('data-src')} | ALT: {img.get('alt')}")

print("\n--- LINK TEXTS ---")
for a in soup.find_all('a'):
    text = a.get_text().strip()
    if text:
        print(f"LINK: {text} -> {a.get('href')}")
